### PR TITLE
Watch PipelineRuns as well

### PR DESCRIPTION
**Pull request by stuartwdouglas**
This will allow us to notify on the success or failure of the jobs

### CODE REVIEW

File: pkg/reconciler/pipelinerun/pipelinerun.go
Comment (gabemontero) [inline pkg/reconciler/pipelinerun/pipelinerun.go:50]: so this is what we discussed last night, so to be clear, first pass this is what I was expecting

but in case either a) I failed to mention this tidbit, or b) it got lost in the torrent of details we went over, even if we continue down this "explicit" reconciler approach (vs. the "shipwright" approach I've been mentioning), we'll most likely want to employ the options available when setting up the controller in the first place so that it filters pipelineruns so we only get events for the ones with our label

The controller runtime doc that tries to describe this is at https://github.com/kubernetes-sigs/controller-runtime/blob/master/designs/use-selectors-at-cache.md

Then in our code, in the `SetupNewReconcilerWithManager` methods, we would switch from the "simpler" `ctrl.NewControllerManagedBy` call to the more "powerfull/flexible" `ctrl.NewManager` call, where we set the `Options.NewCache` function.

@stuartwdouglas ^^

Comment (stuartwdouglas) [reply pkg/reconciler/pipelinerun/pipelinerun.go]: @gabemontero I have added a commit that attempts to add what you describe, although the code is a bit different to what you mentioned so I am not sure if it is right.

Comment (gabemontero) [reply pkg/reconciler/pipelinerun/pipelinerun.go]: ok got a few minutes taking a look now @stuartwdouglas

**Review by gabemontero — CHANGES_REQUESTED** [review]
ok an initial pass @stuartwdouglas 

overall seems on the right path

made a few suggestions; other than the label selector value thing, which might not be needed based on my recollections, but I *think* it might be, but we'll see what your testing says, I think you golang journey is off to a good start :-) 

I'll probably tackle the "shipwright approach consideration" I keep mentioning (which again may not be needed) after this merges, unless you hit some blocker during testing and we have to revisit things.

And full disclosure, and probably not a surprise, I focused on the golang and ignored the java

File: pkg/controller/controller.go
Comment (gabemontero) [reply pkg/controller/controller.go:41]: ok I realized something later tonight about my comment and I took another look, where my prior comments missed a couple of things ... I believe I now noticed that
- there is a 1 to 1 correspondence between artifactbuildrequests and taskruns, the 1 to 1 between dependencybuilds and pipelineruns based on the use of create in the artifactbuidlrequests and dependencybuild reconcilers
- but I see you doing lists in your pipelinerun and taskrun reconcilers to find their taskruns and pipelineruns

If I have that 1 to 1 correspondence right, I think we need to adjust this a little, though not in the way I mentioned in https://github.com/redhat-appstudio/jvm-build-service/pull/33#discussion_r873068039

Essentially, we have two labels:
1) a label with a constant k/v pair (where maybe the empty string for the value is OK, but if not just pick a non empty constant value) to help filter the pipelinerun and taskrun reconcilers to those associated with artifactbuildrequests and dependencybuilds
2) a second label where the value if the namespace/name needed so you use a Get instead of a List in the pipelinerun and taskrun reconcilers

File: pkg/apis/jvmbuildservice/v1alpha1/dependencybuild_types.go
Comment (stuartwdouglas) [reply pkg/apis/jvmbuildservice/v1alpha1/dependencybuild_types.go:26]: I ran it and there were no changes. Looking at the code I guess because it is a primitive type it is handled by the simple copy case?

File: pkg/reconciler/artifactbuildrequest/artifactbuildrequest.go
Comment (stuartwdouglas) [reply pkg/reconciler/artifactbuildrequest/artifactbuildrequest.go]: Done

File: pkg/controller/controller.go
Comment (stuartwdouglas) [reply pkg/controller/controller.go:41]: Are you saying that I should put the name of the ABR in a Label on the TaskRun and use that to lookup the ABR directly? 

The issue with this is that label values are much more restrictive than the naming rules, so the generated ABR names currently don't fit into a value. I could shrink the names somehow and enforce that they meet the label rules, but some human readability may be lost for long GAV's.

My solution here was to generate an artificial ID and put it in a label on both resources, and then use List to update the resources, but I can shrink the names if you want. 

Note that I already have two labels, one for correlation and empty one for the filtering like you suggest.

**Comment by stuartwdouglas** [comment]
@gabemontero I also made a first attempt at adding a test based on the tests in the build service operator. It seems to work but I would like you to have a look at it before I spend any time adding more to make sure I am on the right path.

File: pkg/reconciler/taskrun/suite_test.go
Comment (gabemontero) [reply pkg/reconciler/taskrun/suite_test.go:75]: I do see you made Makefile changes similar to https://github.com/redhat-appstudio/build-service/blob/main/Makefile so perhaps you were able to run locally?  If you mentioned in slack somewhere on in one of our calls and I'm forgetting what you said, apologies.

Comment (gabemontero) [reply pkg/reconciler/taskrun/suite_test.go:75]: But otherwise, I think what you have is a good first step that we can build upon.

Comment (stuartwdouglas) [reply pkg/reconciler/taskrun/suite_test.go:75]: @gabemontero yes, I was able to run locally against minikube and it passed

Comment (gabemontero) [reply pkg/reconciler/taskrun/suite_test.go:75]: ok that is a good start @stuartwdouglas 

now, the openshift/release stuff I noted ^^ runs against app studio deployed against an openshift cluster ... see https://github.com/redhat-appstudio/build-service/tree/main/.ci for example

while I think we would want that at some point, I have worked on projects where they launch kind or minikube from github actions for e2e testing like that as well; not knowing your experience launching / testing on openshift, vs. however painful it might be to get me to a point where I run end to end testing with this project, maybe we employ a github actions / "vanilla" k8s e2e approach for an interim period.

I'll defer to you on whether you want to include launching the e2e from github actions with this PR or a subsequent one (don't see it in https://github.com/redhat-appstudio/jvm-build-service/pull/33/files#diff-aa526adb0a3d690b8bce5b9ae75da3c3a254ebeffe4b108851b60d8746356276 currently)

long term, at first blush, it would not be a "bad thing" to have e2e tests on both vanilla k8s and openshift, though I don't see that approach being taken for the build-service or application-service repos.

**Pull request merged**
